### ISSUE TITLE
Flaky unit test: kubelet.TestFreeSpaceImagesAlsoDoesLookupByRepoTags flakes on Jenkins

### ISSUE THREAD

**Issue by quinton-hoole** [kubernetes/kubernetes#14588]
job/kubernetes-pull-build-test-e2e-gce/10158/testReport/junit/(root)/kubelet/TestFreeSpaceImagesAlsoDoesLookupByRepoTags/

```
kubelet.TestFreeSpaceImagesAlsoDoesLookupByRepoTags (from k8s.io_kubernetes_pkg_kubelet)

Failing for the past 1 build (Since Unstable#10158 )
Took 0 ms.
add description
Error Message

Failed
Stacktrace

<autogenerated>:31: 
                        
	Error Trace:	image_manager_test.go:332
	
	Error:		Not equal: 1024 (expected)
		        != 2048 (actual)
```

**Comment by brendandburns** [comment]
Flaked in Shippable too:

https://app.shippable.com/builds/5605aea47d8ec70c00fd19a7

@dchen1107 @kubernetes/goog-node

**Comment by bgrant0607** [comment]
My guess: #14035

@feiskyer

**Comment by quinton-hoole** [comment]
This just looks like a plain bug to me. Not clear how it ever passed.

```
func TestFreeSpaceImagesAlsoDoesLookupByRepoTags(t *testing.T) {
	manager, fakeRuntime, _ := newRealImageManager(ImageGCPolicy{})
	fakeRuntime.ImageList = []container.Image{
		makeImage(0, 1024),
		{
			ID:   "5678",
			Tags: []string{"potato", "salad"},
			Size: 2048,
		},
	}
	fakeRuntime.ContainerList = []*container.Container{
		{
			ID:    "c5678",
			Image: "salad",
		},
	}

	spaceFreed, err := manager.freeSpace(1024)
	assert := assert.New(t)
	require.NoError(t, err)
	assert.EqualValues(1024, spaceFreed)
	assert.Len(fakeRuntime.ImageList, 1)
}

```

**Comment by bgrant0607** [comment]
I'm going to revert it.

**Comment by bgrant0607** [comment]
PR #14590

**Comment by yujuhong** [comment]
> This just looks like a plain bug to me. Not clear how it ever passed.

@quinton-hoole, the test is mostly correct but the it should've updated the fakeRuntime.PodList (not the ContainerList). It was flaky because sometimes the image manager would remove the the correct image. I'll comment on the original PR.

**Comment by yujuhong** [comment]
[Commented on PR](https://github.com/kubernetes/kubernetes/pull/14035/files#r40485200) about the fix. Also uploaded #14595 to remove unused code to prevent accidental use. The tests could also be improved since there is some chance that the test may just remove the right image, rendering the test meaningless.

**Comment by dchen1107** [comment]
Yeah, in @feisky's PR all other places are using PodList, except this one.

**Comment by feiskyer** [comment]
Fix is at #14599

**Comment by dchen1107** [comment]
#14599 were merged, close the issue.
